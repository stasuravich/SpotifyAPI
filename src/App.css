.Container {
  text-align: center;

}
.Box1 {
  display: flex;
  position: sticky;
  top: 0;
  background: #E8E8E8;
  height: 60px;
  z-index: 10;
}

.Box2{
  display: flex;
}

div{
  margin: 5px;

}
