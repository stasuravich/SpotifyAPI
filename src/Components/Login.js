import React, {useState} from 'react';
import '../Css/Login.css';
import logo from '../logos/adrian-korte-5gn2soeAc40-unsplash.jpg';

const Login =props=>{

  const [token, setToken]=useState();


  const submit=e =>{
    e.preventDefault();
    if(token){
      props.setLoggedIn(token);
    }
    else{
      props.setWrong(true);
    }
  }


  return (
    <div className="Login">
      <img src={logo} alt ="Logo" className="Logo"/>
      <h1>Login</h1>
      <form onSubmit = {e => submit(e)}>
        <label className="TokenLabel">Enter spotify token below (get it from {" "}
          <a href="https://beta.developer.spotify.com/documentation/web-playback-sdk/quick-start/#authenticating-with-spotify">
            here
          </a>):
        </label>
        <br/>
        <input className="TokenInput" type="text" placeholder="spotify token" onChange={e=> setToken(e.target.value)} defaultValue =''/>
        <br/><br/>
        <button type = "submit" >Submit</button>
      </form>
      {props.wrong &&
        <div className="Wrong"> Invalid token, please try again</div>
      }
    </div>
  );
}

export default Login;
