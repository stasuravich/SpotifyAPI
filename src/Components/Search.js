import React, {useRef, memo} from 'react';
import '../Css/Search.css';

const Search = memo(function Search(props){
  //console.log(props.class);
  const inputRef = useRef();

  props.alerter(inputRef);

  return(
    <div className= {props.class} >
      <input ref = {inputRef} type="text" placeholder = {props.placeh} onChange={(e) => props.setQuery(e.target.value)}></input>
    </div>
  )
})

export default Search;
