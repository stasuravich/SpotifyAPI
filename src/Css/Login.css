.Login{
  text-align: center;
}
.Logo{
  padding-top: 25px;
  width: 15%;
}

.TokenLabel{
  display: block;
  font-size: 20px;
}
.TokenInput{
  width: 300px;
}

.Wrong{
  color: red;
}
