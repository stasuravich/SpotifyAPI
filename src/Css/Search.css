.SearchPlaylist{
  padding-top: 10px;
  order: 1;
  padding-left: 250px;
  visibility: hidden;
}

.SearchOnline{
  padding-top: 10px;
  order: 3;
  padding-right: 250px;
}
