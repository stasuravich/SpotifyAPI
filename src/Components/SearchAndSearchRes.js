import React, {useState, useRef, memo, useCallback} from 'react';
import Online from './Online';
import Search from './Search';

import axios from 'axios';

const SearchAndSearchRes = memo(props=>{
  const [searchRes, setSearchRes] = useState();
  const inputRef = useRef();

  const changeSearchRes = useCallback(query =>{
    if(query){
      axios(`https://api.spotify.com/v1/search?limit=15&type=track&q=${query}`, {
          method: 'GET',
          headers: {'Authorization' : 'Bearer ' + props.logInfo}
        })
        .then((response) => {
          if(response.data.tracks.items.length){
            setSearchRes(response.data);
          }
          else{
            setSearchRes(query);
          }
        })
    }
    else{
      setSearchRes(null);
    }
  }, [searchRes])

  return(
    <>
      <Search class="SearchOnline" placeh="Search for a song..." inputRef={inputRef} changeSearchRes={changeSearchRes}/>
      <Online songs={searchRes} logInfo={props.logInfo} playlists = {props.playlists} getPlaylist={props.getPlaylist} playlistTrack={props.playlistTrack} setTracks={props.setTracks} tracks={props.tracks} setDispPlaylist={props.setDispPlaylist} inputRef={inputRef} />
    </>
  )
})

export default SearchAndSearchRes;
