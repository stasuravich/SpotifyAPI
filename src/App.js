import React, {useState, useEffect, useRef} from 'react';
import './App.css';
//import SpotifyWeb from 'spotify-web-api-js';
import Playlists from './Components/Playlists';
import Songs from './Components/Songs';
import Details from './Components/Details';
import Search from './Components/Search';
import Login from './Components/Login';
import SearchAndSearchRes from './Components/SearchAndSearchRes';

import axios from 'axios';

//var spotifyWebApi = new SpotifyWeb();

function App() {
  //console.log("In App");
  const [logInfo, setLogInfo]=useState('');
  const [wrong, setWrong]=useState(false);
  const [playlists, setPlaylists] = useState({selectedPlaylist:'', listOfPlaylistsFromAPI: []});
  const [dispPlaylist, setDispPlaylist] = useState();
  const [tracks, setTracks] = useState({selectedTrack:'', listOfTracksFromApi: ''});
  const [qPlaylist, setQPlaylist] = useState('');
  const playlistTrack=useRef();
  //const params = getHashParams();

  useEffect(()=> {
    if (logInfo){
      //spotifyWebApi.setAccessToken(logInfo);
      axios('https://api.spotify.com/v1/me/playlists', {
          method: 'GET',
          headers: {'Authorization' : 'Bearer ' + logInfo}
        })
        .then((response) => {
          setPlaylists({listOfPlaylistsFromAPI: response.data.items});
        })
        .catch(_=>{
            setWrong(true);
          })
    }
  }, [logInfo]);


  // function getHashParams() {
  //   var hashParams = {};
  //   var e, r = /([^&;=]+)=?([^&;]*)/g,
  //       q = window.location.hash.substring(1);
  //   while ( e = r.exec(q)) {
  //      hashParams[e[1]] = decodeURIComponent(e[2]);
  //   }
  //   return hashParams;
  // }
  const getPlaylist= async (offset, curTracks, val)=>{
    await axios(`https://api.spotify.com/v1/playlists/${val}/tracks?offset=${offset}`, {
      method: 'GET',
      headers: {'Authorization' : 'Bearer ' + logInfo}
    })
    .then (response => {
      if(curTracks){
        curTracks.push(...response.data.items);
      }
      else{
        curTracks=response.data.items;
      }
      if(response.data.items.length===100){
        curTracks= getPlaylist(offset+100, curTracks, val);
      }
    })
    return curTracks;
  };

  return (
    <div className="App">
      {!playlists.listOfPlaylistsFromAPI.length ? <Login setLoggedIn={setLogInfo} wrong={wrong} setWrong={setWrong}/>:
      <>
        <Search class="SearchPlaylist" placeh="Search playlist" setQuery={setQPlaylist}/>
        <SearchAndSearchRes logInfo={logInfo} playlists={playlists} getPlaylist={getPlaylist} playlistTrack={playlistTrack} setTracks={setTracks} tracks={tracks} setDispPlaylist={setDispPlaylist}/>
        <Playlists playlists= {playlists} setDispPlaylist={setDispPlaylist} getPlaylist={getPlaylist} setPlaylists={setPlaylists} />
        <div className="Box">
          {dispPlaylist && <Songs logInfo={logInfo} playlists={playlists} tracks={tracks} setTracks = {setTracks} playlistTrack={playlistTrack} dispPlaylist={dispPlaylist} setDispPlaylist={setDispPlaylist} query={qPlaylist} getPlaylist={getPlaylist}/> }
          {tracks.selectedTrack && <Details access_token={logInfo} tracks={tracks} setTracks={setTracks} playlistTrack={playlistTrack}/>}
        </div>
      </>}
    </div>
  );
}
export default App;
