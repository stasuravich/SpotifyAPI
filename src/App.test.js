import React, {createRef} from 'react';
import App from './App';
import Login from './Components/Login';
import Search from './Components/Search';
import SearchAndSearchRes from './Components/SearchAndSearchRes';
//import Playlists from './Components/Playlists';
import {shallow, mount} from "enzyme";
import toJson from "enzyme-to-json";

const credWrong="wrong";
const playlistClass="SearchPlaylist";
const logInfo="token";
//const playlists=createRef({selectedPlaylist:'EDM', listOfPlaylistsFromAPI: [{"EDM"}, {"Country"}, {"Pop"}]});

describe("rendering components", ()=> {
  it("renders App without crashing", ()=>{
    shallow(<App/>);
  });
  it("renders Login component without crashing", ()=>{
    shallow(<Login/>);
  })
})

describe("passing props", ()=>{
  const loginWrapper=mount(<Login cred={credWrong} />);
  const searchWrapper=mount(<Search class={playlistClass}/>);
  const searchResWrapper=mount(<SearchAndSearchRes logInfo={logInfo}/>)
  //const playlistsWrapper=mount(<Playlists playlists={playlists}/>)
  it("accepts Login props", () => {
    expect(loginWrapper.props().cred).toEqual(credWrong);
  })
  it("notification contains correct value", ()=>{
    const value=loginWrapper.find(".Wrong").text();
    const expectdeValue="Invalid token, please try again";
    expect(value).toEqual(expectdeValue);
  })
  it("accepts Search prop", ()=>{
    expect(searchWrapper.props().class).toEqual(playlistClass);
  })
  it("accepts SearchRes prop", ()=>{
    expect(searchResWrapper.props().logInfo).toEqual(logInfo);
  })
})

describe("snapshots", () => {
  it("App snapshots", () => {
    const tree= shallow(<App/>);
    expect(toJson(tree)).toMatchSnapshot();
  });
})
