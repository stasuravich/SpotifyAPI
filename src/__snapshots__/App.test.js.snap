// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots App snapshots 1`] = `
<div
  className="App"
>
  <Memo()
    setCred={[Function]}
    setLoggedIn={[Function]}
  />
</div>
`;
